Author: Nicholas Bamber <nicholas@periapt.co.uk>
Subject: Stop makefiles from overwriting themselves
 Upstream uses an unconventional makefile strategy in which
 makefiles copy makefiles (and one header) around to cater for
 compatibility. This patch overrides build/Makefile.linux to
 stop that, and replicates the effects of the upstream code
 by patching some Makefiles to the same effect.
 The debian/rules also plays its part by backing and restoring 
 the header file and by explcitly mentioning Makefiles as required.
Forwarded: not-needed
Last-Update: 2011-09-09
--- a/build/Makefile.linux
+++ b/build/Makefile.linux
@@ -53,12 +53,10 @@
 
 clean:
 	rm -f $(OBJECTS) core $(EXECS) ; \
-	cp build/Makefile.w Makefile ; cd dns ; make clean ; \
+	cd dns ; make clean ; \
 	cd ../libs ; make clean ; cd ../parse ; make clean ; \
 	cd ../qual ; make clean ; \
 	cd ../server ; make clean ; \
-	cp Makefile.recursive Makefile ; \
-	cd ../test ; make clean ; \
 	cd ../tools ; make clean ; \
 	cd misc ; make clean ; \
 	cd ../../utf8 ; make clean ; \
--- a/build/Makefile.freebsd
+++ b/build/Makefile.freebsd
@@ -30,17 +30,17 @@
 M="VERSION=$(VERSION)"
 Q="DEFINES=-DSELECT_PROBLEM"
 
+FLAGS = -O2 -Wall   -pipe -D_THREAD_SAFE -pthread
+#FLAGS = -g
+
 # FreeBSD needs some special flags to compile MaraDNS
-CC="cc -O2 -Wall -pipe -D_THREAD_SAFE -pthread"
+CC="cc $(FLAG)"
 M=CC=$(CC) 
 V="VERSION=$(VERSION)"
 
 # Debug
 D=$(M) DEBUG=-DDEBUG
 
-FLAGS = -O2 -Wall 
-#FLAGS = -g
-
 all: 
 	cd libs ; make $(M) ; cd ../dns ; make $(M) ; \
 	cd ../rng ; make $(M) ; cd ../parse ; make $(M) ; \
